subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -331,19 +364,44 @@ void ll_filters_scan_update(u8_t scan_fp)
 
 int ll_rl_find(u8_t id_addr_type, u8_t *id_addr)
 {
-	int i, free = -IDX_NONE;
+	int i, free = -RL_IDX_NONE;
 
 	for (i = 0; i < CONFIG_BLUETOOTH_CONTROLLER_RL_SIZE; i++) {
 		if (LIST_MATCH(rl, i, id_addr_type, id_addr)) {
 			return i;
-		} else if (!rl[i].taken && free == -IDX_NONE) {
+		} else if (!rl[i].taken && free == -RL_IDX_NONE) {
 			free = -i;
 		}
 	}
 
 	return free;
 }
 
+bool ctrl_rl_allowed(u8_t id_addr_type, u8_t *id_addr)
+{
+	int i, j;
+
+	if (!rl_enable) {
```

Comment:
Remove this and use ctrl_rl_enabled() in caller. Caller can then short-circuit passing the two parameters, saves some cycles.